src/rsa/rsa.rs

```diff
@@ -63,17 +67,19 @@ impl PositiveInteger {
         }
         // Reject leading zeros. Also reject the value zero ([0]) because zero
         // isn't positive.
-        if untrusted::Reader::new(input).peek(0) {
+        let top_byte =  try!(untrusted::Reader::new(input).read_byte());
+        if top_byte == 0 {
             return Err(error::Unspecified);
         }
+        let bit_len = input.len() * 8 - top_byte.leading_zeros() as usize;
```

Comment:
`input.len() * 8` could overflow. We need to prevent that.

Maybe we should just calculate/store the number of leading zeros in the top byte directly, instead of calculating the length of the modulus in bits?